bluephos/modules/octahedral_embed.py

```diff
@@ -0,0 +1,178 @@
+    Chem.SanitizeMol(outmol)
+
+    return outmol
+
+
+def transfer_conformation(mol, substruct, conformer=0):
+    """Given a molecule, and a second molecule which is a substructure of the
+    first, assign coordinates to the substructure based on the matching part of
+    the original molecule"""
+    match = mol.GetSubstructMatch(substruct)
+    substruct_conformation = rdchem.Conformer(substruct.GetNumAtoms())
+    for i, index in enumerate(match):
+        point = mol.GetConformer(conformer).GetAtomPosition(index)
+        substruct_conformation.SetAtomPosition(i, point)
+    substruct.AddConformer(substruct_conformation)
+
+
+def run_three_times(mol, reaction):
+    for i in range(3):
+        mol = reaction.RunReactants([mol])[0][0]
+    return mol
+
+
+def get_directory_path():
+    """Returns the directory path of the current script."""
+    return os.path.dirname(os.path.abspath(__file__))
+
+
+def skeleton_extraction(mol, template):
+    """Extracts and returns the skeleton of a molecule based on the provided SMARTS template."""
+    matches = mol.GetSubstructMatches(template)
+    matching_indices = set(sum(matches, ()))
+    editable_mol = rdchem.RWMol(mol)
+    editable_mol.BeginBatchEdit()
+    for atom in editable_mol.GetAtoms():
+        if atom.GetIdx() not in matching_indices:
+            editable_mol.RemoveAtom(atom.GetIdx())
+        atom.SetFormalCharge(0)
+    editable_mol.CommitBatchEdit()
+    return editable_mol.GetMol()
+
+
+def load_molecule(file_name):
+    """Loads a molecule from a .mol2 file given a file name."""
+    dir_path = get_directory_path()
+    base_mol = rdmolfiles.MolFromMol2File(os.path.join(dir_path, file_name))
+    dative_mol = make_bonds_dative(base_mol)
+    rdmolops.RemoveStereochemistry(dative_mol)
+    return dative_mol
+
+
+def compute_skeletons(isomer):
+    """Computes and returns all skeletons based on the isomer type."""
+
+    if isomer == "fac":
+        base_mol = load_molecule("OHUZEW.mol2")
+        carbene_mol = load_molecule("MAXYIU.mol2")
+    elif isomer == "mer":
+        base_mol = load_molecule("OHUZIA.mol2")
+        carbene_mol = load_molecule("MAXYOA.mol2")
+    else:
+        raise ValueError(f'Isomer should be "mer" or "fac", given {isomer}')
+
+    template = rdmolfiles.MolFromSmarts("[Ir]1~n:[*]~[*]:c~1")
+
+    skeleton = skeleton_extraction(base_mol, template)
+
+    carbene_skeleton = rdmolfiles.MolFromSmarts(CARBENE_SKELETON_SMARTS)
+    transfer_conformation(carbene_mol, carbene_skeleton)
+    reactions = [
+        rdChemReactions.ReactionFromSmarts(
+            "[Ir:1]1<-[n:2]:[n:3]~[c:4]:[c:5]~1>>[Ir:1]1<-[n:2]:[c:3]~[n:4]:[c:5]~1"
+        ),
+        rdChemReactions.ReactionFromSmarts(
+            "[Ir:1]1<-[n:2]:[n:3]~[c:4]:[c:5]~1>>[Ir:1]1<-[n:2]:[n:3]~[n:4]:[c:5]~1"
+        ),
+        rdChemReactions.ReactionFromSmarts(
+            "[Ir:1]1<-[n:2]:[n:3]~[c:4]:[c:5]~1>>[Ir:1]1<-[n:2]:[c:3]~[c:4]:[c:5]~1"
+        ),
+    ]
```

Comment:
Might be useful that each of these are also a module var. And could be good to highlight the difference which took me a while to find! Like:

```py
REACTION_PREFIX = "[Ir:1]1<-[n:2]:[n:3]~[c:4]:[c:5]~1>>[Ir:1]1<-[n:2]:"
REACTION_SUFFIX = ":[c:5]~1"

reactions = [
    ReactionFromSmarts(f'{REACTION_PREFIX}core{REACTION_SUFFIX}') 
    for i in ['[c:3]~[n:4]', '[n:3]~[n:4]', '[c:3]~[c:4]']
]
```

Reply:
Nice solution!